openmeetings-service/src/main/java/org/apache/openmeetings/service/calendar/caldav/iCalUtils.java

```diff
@@ -0,0 +1,418 @@
+                if(uri.getScheme().equals("mailto")){
+                    String email = uri.getSchemeSpecificPart();
+                    User u = userDao.getByEmail(email);
+                    if(u == null)
+                        u = userDao.getContact(email, a.getOwner());
+                    attList.add(createMeetingMember(a, u));
+                }
+            }
+        }
+
+        a.setMeetingMembers(attList.isEmpty() ? null : attList);
+
+        return a;
+    }
+
+    private MeetingMember createMeetingMember(Appointment a, User u){
+        MeetingMember mm = new MeetingMember();
+        mm.setUser(u);
+        mm.setDeleted(false);
+        mm.setInserted(a.getInserted());
+        mm.setUpdated(a.getUpdated());
+        mm.setAppointment(a);
+        return mm;
+    }
+
+    private Room createDefaultRoom() {
+        Room r = new Room();
+        r.setAppointment(true);
+        if (r.getType() == null) {
+            r.setType(Room.Type.conference);
+        }
+        return r;
+    }
+
+    public TimeZone parseTimeZone(Calendar calendar, User owner){
+        if(calendar != null) {
+            Component timezone = calendar.getComponent(Component.VTIMEZONE);
+            if (timezone != null) {
+                Property tzid = timezone.getProperty(Property.TZID);
+                if (tzid != null)
+                    return timezoneUtil.getTimeZone(tzid.getValue());
+            }
+        }
+        return timezoneUtil.getTimeZone(owner);
+    }
+
+    /**
+     * Convenience function to parse date from {@link net.fortuna.ical4j.model.Property} to
+     * {@link Date}
+     * @param dt DATE-TIME Property from which we parse.
+     * @param timeZone Timezone of the Date.
+     * @return {@link java.util.Date} representation of the iCalendar value.
+     */
+    public Date parseDate(Property dt, TimeZone timeZone) {
+        if(dt == null || dt.getValue().equals(""))
+            return null;
+
+        String[] acceptedFormats = { "yyyyMMdd'T'HHmmss", "yyyyMMdd'T'HHmmss'Z'", "yyyyMMdd"};
+        Parameter tzid = dt.getParameter(Parameter.TZID);
+        if(tzid == null)
+            return parseDate(dt.getValue(), acceptedFormats, timeZone);
+        else
+            return parseDate(dt.getValue(), acceptedFormats, timezoneUtil.getTimeZone(tzid.getValue()));
+    }
+
+    /**
+     * Adapted from DateUtils to support Timezones, and parse ical dates into {@link java.util.Date}.
+     * Note: Replace FastDateFormat to java.time, when shifting to Java 8 or higher.
+     * @param str Date representation in String.
+     * @param patterns Patterns to parse the date against
+     * @param timeZone Timezone of the Date.
+     * @return <code>java.util.Date</code> representation of string or
+     * <code>null</code> if the Date could not be parsed.
+     */
+    public Date parseDate(String str, String[] patterns, TimeZone timeZone) {
+        FastDateFormat parser = null;
```

Comment:
Because it's faster and is ThreadSafe. With SimpleDateFormat, both of those are at a disadvantage.